super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,31 +231,65 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
-				if( c == quoteChar ) {
+
+				if( c == quoteEscapeChar ) {
 					int nextCharIndex = charIndex + 1;
 					boolean availableCharacters = nextCharIndex < line.length();
 					boolean nextCharIsQuote = availableCharacters && line.charAt(nextCharIndex) == quoteChar;
+					boolean nextCharIsEscapeQuoteChar = availableCharacters && line.charAt(nextCharIndex) == quoteEscapeChar;
+
 					if( nextCharIsQuote ) {
 						/*
-						 * An escaped quote (""). Add a single quote, then move the cursor so the next iteration of the
-						 * loop will read the character following the escaped quote.
+						 * An escaped quote (e.g. "" or \"). Skip over the escape char, and add
+						 * the following quote char as part of the column;
+						 */
+						charIndex++;
+						currentColumn.append(line.charAt(nextCharIndex));
+					} else if( nextCharIsEscapeQuoteChar ) {
+						/*
+						 * A double escape (normally \\). Save the escape char, then continue to
+						 * next character.
 						 */
 						currentColumn.append(c);
 						charIndex++;
-						
-					} else {
+					} else if( quoteEscapeChar == quoteChar ) {
 						/*
-						 * A single quote ("). Update to NORMAL (but don't save quote), then continue to next character.
+						 * If the escape char is also the quote char and we didn't escape a
+						 * subsequent character, then this is a lone quote and the end of the
+						 * field.
 						 */
 						state = TokenizerState.NORMAL;
 						quoteScopeStartingLine = -1; // reset ready for next multi-line cell
+					} else {
+						/*
+						 * Escape char wasn't before either another escape char or a quote char,
+						 * so process it normally.
+						 */
+						currentColumn.append(c);
+					}
+				} else if( c == quoteChar ) {
+
+					/*
+					 * A single quote ("). Update to NORMAL (but don't save quote), then continue to next character.
+					 */
+					state = TokenizerState.NORMAL;
+					quoteScopeStartingLine = -1; // reset ready for next multi-line cell
+
+					int nextCharIndex = charIndex + 1;
+					boolean availableCharacters = nextCharIndex < line.length();
+					boolean nextCharIsQuote = availableCharacters && line.charAt(nextCharIndex) == quoteChar;
+
+					if( quoteEscapeChar != quoteChar && nextCharIsQuote ) {
```

Comment:
It would often turn into a multi-line quote and throw the exception from above:

```
						String msg = maxLinesPerRow == 1 ?
								String.format("unexpected end of line while reading quoted column on line %d",
											  getLineNumber()) :
								String.format("max number of lines to read exceeded while reading quoted column" +
											  " beginning on line %d and ending on line %d",
											  quoteScopeStartingLine, getLineNumber());
						throw new SuperCsvException(msg);
```

What's an example of a string where the behavior changed?  I can write a test for it so we can verify.